Runtime/Scripts/Read3DTileset.cs

```diff
@@ -87,9 +87,19 @@ public class Read3DTileset : MonoBehaviour
         [HideInInspector] public UnityEvent<UnityWebRequest.Result> OnServerRequestFailed = new();
         [HideInInspector] public UnityEvent<ContentMetadata> OnLoadAssetMetadata = new();
 
-
+        public string CredentialQuery { get; private set; } = string.Empty;
+        
+        public void ClearKeyFromURL()
+        {
+            if (CredentialQuery != string.Empty)
+            {
+                tilesetUrl = tilesetUrl.Replace(CredentialQuery, string.Empty);
+            }
+        }
+        
         public void ConstructURLWithKey()
         {
+            ClearKeyFromURL(); //remove existing key if any is there
```

Comment:
the problem now is that if you add multiple valid keys, it will create a query with ?key=myKey1&key=myKey2, perhaps a function should be added to read the key from the user's url?

Reply:
And if you move this inside the if statements below? That you only reset it then public or personal key is set?

Reply:
I didn't see that this method must be explicitly called and is not called every time a URL is passed, so: nothing to see here as well!